lib/library.cpp

```diff
@@ -694,8 +694,7 @@ Library::Error Library::loadFunction(const tinyxml2::XMLElement * const node, co
         } else if (functionnodename == "pure")
             func.ispure = true;
         else if (functionnodename == "const") {
-            func.ispure = true;
-            func.isconst = true; // a constant function is pure
```

Comment:
as I understand a const function is always pure. does valueflow assume that the function reads from global memory if it's pure?

Reply:
I'm not sure how ValueFlow uses these attributes.
We also need to figure out what `const` is supposed to mean in the library, see https://github.com/danmar/cppcheck/pull/6385